extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnRunTimeConfig.java

```diff
@@ -0,0 +1,157 @@
+package io.quarkus.security.webauthn;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.vertx.ext.auth.webauthn.Attestation;
+import io.vertx.ext.auth.webauthn.AuthenticatorAttachment;
+import io.vertx.ext.auth.webauthn.AuthenticatorTransport;
+import io.vertx.ext.auth.webauthn.PublicKeyCredential;
+import io.vertx.ext.auth.webauthn.UserVerification;
+
+@ConfigRoot(name = "webauthn", phase = ConfigPhase.RUN_TIME)
+public class WebAuthnRunTimeConfig {
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<String> origin;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<String> domain;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "USB,NFC,BLE,INTERNAL")
+    public Optional<List<AuthenticatorTransport>> transports;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public RelyingPartyConfig relyingParty;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<AuthenticatorAttachment> authenticatorAttachment;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "false")
+    public Optional<Boolean> requireResidentKey;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "REQUIRED")
+    public Optional<UserVerification> userVerification;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "60s")
+    public Optional<Duration> timeout;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "NONE")
+    public Optional<Attestation> attestation;
+
+    // Needs to be a list, order is important
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "ES256,RS256")
+    public Optional<List<PublicKeyCredential>> pubKeyCredParams;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "64")
+    public OptionalInt challengeLength;
+
+    // FIXME
+    //private JsonObject extensions;
+
+    // FIXME
+    //private Map<String, X509Certificate> rootCertificates;
+
+    // FIXME
+    //private List<X509CRL> rootCrls;
```

Comment:
This one is like the previous. Without this, MDS doesn't work. MDS is important for PSD2.0 apps (banking/finance) where authenticators must be verified, for example, a bank should not trust a rooted android, as a bad actor could attempt to extract the private key from the hardware in case of security bugs.

Reply:
Same as before, we can propose an upstream update that takes a list of files.